model/standalone_driver/src/icon4py/model/standalone_driver/standalone_driver.py

```diff
@@ -0,0 +1,618 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import datetime
+import functools
+import logging
+import pathlib
+from collections.abc import Callable
+
+import gt4py.next as gtx
+from gt4py.next import config as gtx_config, metrics as gtx_metrics
+
+import icon4py.model.common.utils as common_utils
+from icon4py.model.atmosphere.diffusion import diffusion, diffusion_states
+from icon4py.model.atmosphere.dycore import dycore_states, solve_nonhydro as solve_nh
+from icon4py.model.common import dimension as dims, model_backends, model_options, type_alias as ta
+from icon4py.model.common.decomposition import definitions as decomposition_defs
+from icon4py.model.common.grid import grid_manager as gm, gridfile, vertical as v_grid
+from icon4py.model.common.initialization import topography
+from icon4py.model.common.metrics import metrics_attributes as metrics_attr
+from icon4py.model.common.states import prognostic_state as prognostics
+from icon4py.model.common.utils import data_allocation as data_alloc, device_utils
+from icon4py.model.standalone_driver import config as driver_config, driver_states, driver_utils
+
+
+log = logging.getLogger(__name__)
+
+
+class Icon4pyDriver:
+    def __init__(
+        self,
+        config: driver_config.DriverConfig,
+        backend: model_backends.BackendLike,
+        grid_manager: gm.GridManager,
```

Comment:
Do you need a reference to the `GridManager` : To me that was always thought of a tool that sets up some functionality at the very initial state of the model run and then "goes out of scope" and could be `del` eted, because once you have the `Grid` and the `Geometry` (and the `DecompositionInfo`) it has done all its duty. (That is a bit different from the responsibility of what @vectorflux  means when he talks about the grid_manager for ICON, which is our `GridManager` + `Grid` + `Geometry` in one thing.